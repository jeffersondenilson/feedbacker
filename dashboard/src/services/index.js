import axios from 'axios'
import AuthService from './auth'
import UsersService from './user'
import router from '../router'
import { setGlobalLoading } from '../store/global'
// TODO: criar variaveis de ambiente
const API_ENVS = {
  production: '',
  development: '',
  local: 'http://localhost:3000'
}

const httpClient = axios.create({
  baseURL: API_ENVS.local
})

httpClient.interceptors.request.use(config => {
  setGlobalLoading(true)
  // inclui o token a cada requisição
  const token = window.localStorage.getItem('token')

  if (token) {
    config.headers.common.Authorization = `Bearer ${token}`
  }

  return config
})

const handleError = (error) => {
  // manda erro 500 para o catch
  const canThrowAnError = error.request.status === 0 ||
    error.request.status === 500

  if (canThrowAnError) {
    setGlobalLoading(false)
    throw new Error(error.message)
  }

  if (error.request.status === 401) {
    // TODO: remover token?
    router.push({ name: 'Home' })
  }

  setGlobalLoading(false)
  return error
}

httpClient.interceptors.response.use((response) => {
  setGlobalLoading(false)
  return response
}, handleError)

export default {
  auth: AuthService(httpClient),
  users: UsersService(httpClient)
}
