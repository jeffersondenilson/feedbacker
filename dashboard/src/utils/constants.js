export default {
  LABELS: {
    all: 'Todos',
    issue: 'Problemas',
    idea: 'Ideias',
    other: 'Outros'
  },
  COLORS: {
    all: 'brand-info',
    issue: 'brand-danger',
    idea: 'brand-warning',
    other: 'brand-graydark'
  }
}
