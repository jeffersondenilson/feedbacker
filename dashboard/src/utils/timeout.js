// TODO: teste timeout
export function wait (timeMs) {
  return new Promise(resolve => {
    setTimeout(resolve, timeMs)
  })
}
