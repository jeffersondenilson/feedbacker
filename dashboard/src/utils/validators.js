export function validateEmptyAndLength6 (value) {
  if (!value) {
    return '*Este campo é obrigatório'
  }

  if (value.length < 6) {
    return '*No mínimo 6 caracteres'
  }

  return true
}

export function validateEmptyAndEmail (value) {
  if (!value) {
    return '*Este campo é obrigatório'
  }

  const isValid = /^[a-z0-9._-]+@[a-z0-9]+\.[a-z]+(\.[a-z]+)?$/i.test(value)
  if (!isValid) {
    return '*Este campo precisa ser um e-mail válido'
  }

  return true
}
