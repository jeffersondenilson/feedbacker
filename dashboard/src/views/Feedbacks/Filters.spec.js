import { defineComponent, h, Suspense } from 'vue'
import { mount, flushPromises } from '@vue/test-utils'
import Filters from './Filters'
import services from '@/services'
import { LABELS, COLORS } from '@/utils/constants'

jest.mock('@/services', () => {
  const summary = { all: 7, issue: 3, idea: 3, other: 1 }
  return {
    feedbacks: {
      getSummary: async () => ({ data: summary })
    }
  }
})

// necessÃ¡rio para rodar async setup
const wrapInSuspense = async (component, options) => {
  const wrapper = mount(defineComponent({
    render () {
      // return h(Suspense, null, {
      //   default: h(component),
      //   fallback: h('div', 'fallback')
      // })
      return h(
        'div',
        h(Suspense, null, {
          default: h(component),
          fallback: h('div', 'fallback')
        })
      )
    }
  }), options)

  await flushPromises()
  return wrapper
}

describe('<Filters />', () => {
  it('should render filters correctly', async () => {
    const wrapper = await wrapInSuspense(Filters, {
      rendersDefaultSlot: true
    })

    const filters = wrapper.findAll('[data-test=filter]')
    expect(filters.length).toBe(4)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it.skip('should set issue as active', async () => {
    const wrapper = await wrapInSuspense(Filters, {
      rendersDefaultSlot: true
    })

    // clicar em issue
    // verificar se 'select' foi emitido com type
    // verificar se filter tem as classes de active
    // verificar se filter-amount tem classes de active
    const issueFilter = wrapper.find('[data-test-filter-type=issue]')
    await issueFilter.trigger('click')
    // { click: [ [ [MouseEvent] ] ] }
    console.log('emitted', wrapper.emitted())
    expect(wrapper.emitted()).toHaveProperty('select')
  })
})
