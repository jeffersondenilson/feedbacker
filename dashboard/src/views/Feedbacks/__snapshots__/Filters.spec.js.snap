// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters /> should render filters correctly 1`] = `
<div>
  <div class="flex flex-col">
    <h1 class="text-2xl font-regular text-gray-800"> Filtros </h1>
    <ul class="flex flex-col mt-3 list-none">
      <li class="bg-gray-200 bg-opacity-50 flex items-center justify-between px-4 py-1 rounded cursor-pointer" data-test="filter">
        <div class="flex items-center"><span class="bg-brand-info inline-block w-2 h-2 mr-2 rounded-full"></span> Todos</div><span class="text-brand-info font-bold" data-test="filter-amount">7</span>
      </li>
      <li class="flex items-center justify-between px-4 py-1 rounded cursor-pointer" data-test="filter" data-test-filter-type="issue">
        <div class="flex items-center"><span class="bg-brand-danger inline-block w-2 h-2 mr-2 rounded-full"></span> Problemas</div><span class="text-brand-gray font-bold" data-test="filter-amount">3</span>
      </li>
      <li class="flex items-center justify-between px-4 py-1 rounded cursor-pointer" data-test="filter" data-test-filter-type="idea">
        <div class="flex items-center"><span class="bg-brand-warning inline-block w-2 h-2 mr-2 rounded-full"></span> Ideias</div><span class="text-brand-gray font-bold" data-test="filter-amount">3</span>
      </li>
      <li class="flex items-center justify-between px-4 py-1 rounded cursor-pointer" data-test="filter" data-test-filter-type="other">
        <div class="flex items-center"><span class="bg-brand-graydark inline-block w-2 h-2 mr-2 rounded-full"></span> Outros</div><span class="text-brand-gray font-bold" data-test="filter-amount">1</span>
      </li>
    </ul>
  </div>
</div>
`;
